abci/server/socket_server.go

```diff
@@ -151,6 +151,8 @@ func (s *SocketServer) handleRequests(closeConn chan error, conn net.Conn, respo
 	defer func() {
 		// make sure to recover from any app-related panics to allow proper socket cleanup
 		panicVal = recover()
+		closeConn <- fmt.Errorf("recovered from panic: %v", panicVal)
```

Comment:
you forgot the `if panicVal != nil` part

Reply:
yep, figured it out as I reviewed the code, so it's all good now!